lib/THC/generic/THCTensorMath.cu

```diff
@@ -391,4 +391,42 @@ accreal THCTensor_(trace)(THCState *state, THCTensor *src_) {
   return trace;
 }
 
+#if defined(THC_REAL_IS_FLOAT) || defined(THC_REAL_IS_DOUBLE) || defined(THC_REAL_IS_HALF)
+
+void THCTensor_(linspace)(THCState *state, THCTensor *r_, real a, real b, long n) {
+  THCAssertSameGPU(THCTensor_(checkGPU)(state, 1, r_));
```

Comment:
Sure! In main Torch7 repo we explicitly required that linspace and logspace also works for non-contiguous Tensors in both code and tests. So just want to make sure, we make it behave differently in CUDA backend?